### PR TITLE
Enable auto-uploading/remote-auto-saving for locally changed pages

### LINKED ISSUE #11126: AutoUpload pages with local changes
`UploadStarter` currently enqueues only posts with local changes, but ignores pages. Goal of this task is to start enqueueing even pages with local changes to the `UploadService`. 

Note: We supported this before we released the feature. I think we basically just want to revert 0a123667f9a2bccebc199feed9c1fcabc7afcc2a.

### PR DESCRIPTION
**Pull request by malinajirka**
Fixes #11126 

Enables automatic upload/remote-auto-save for pages with local changes. This feature was disabled before we released Offline Posting. This PR technically just reverts https://github.com/wordpress-mobile/WordPress-Android/commit/0a123667f9a2bccebc199feed9c1fcabc7afcc2a with minor modifications. It also introduces a new tracking event.

To test:
1. Go to Pages List
2. Click on an existing draft
3. Turn on airplane mode
4. Click on Publish
5. Turn off airplane mode
6. Notice the page gets auto-uploaded (check it's published on the blog/calypso)
7. Verify `auto_upload_page_invoked` was tracked with UPLOAD action
------------------------
1. Go to Pages List
2. Click on an existing draft
3. Turn on airplane mode
4. Make a change to the content
5. Press back (don't publish it)
5. Turn off airplane mode
6. Notice the page gets remote-auto-saved (check an unpublished revision is visible in Calypso)
7. Verify `auto_upload_page_invoked` was tracked with REMOTE_AUTO_SAVE action
-----------
We'll do thorough tests before we merge this whole feature into `develop`. So there is no need to perform any deep tests.

PR submission checklist:

- [x] I have considered adding unit tests where possible.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/uploads/UploadStarterTest.kt [hunk 1]
```diff
@@ -93,8 +118,14 @@ class UploadStarterTest {
...
 
+    private val pageStore = mock<PageStore> {
+        sites.forEach {
+            onBlocking { getPagesWithLocalChanges(eq(it)) } doReturn sitesAndDraftPages.getValue(it)
+        }
+    }
+
     @Test
-    fun `when the internet connection is restored and the app is in foreground, it uploads locally changed posts`() {
+    fun `when the internet connection is restored and the app is in foreground, it uploads changed posts&pages`() {
```
Comment (jd-alexander) [inline WordPress/src/test/java/org/wordpress/android/ui/uploads/UploadStarterTest.kt]:
```suggestion
    fun `when the internet connection is restored and the app is in foreground, it uploads changed posts & pages`() {
```

File: WordPress/src/test/java/org/wordpress/android/ui/uploads/UploadStarterTest.kt [hunk 2]
```diff
@@ -144,7 +175,7 @@ class UploadStarterTest {
     }
 
     @Test
-    fun `when the app is placed in the foreground, it uploads locally changed posts`() {
+    fun `when the app is placed in the foreground, it uploads locally changed posts&pages`() {
```
Comment (jd-alexander) [inline WordPress/src/test/java/org/wordpress/android/ui/uploads/UploadStarterTest.kt]:
```suggestion
    fun `when the app is placed in the foreground, it uploads locally changed posts & pages`() {
```

File: WordPress/src/test/java/org/wordpress/android/ui/uploads/UploadStarterTest.kt [hunk 3]
```diff
@@ -158,15 +189,15 @@ class UploadStarterTest {
...
+        verify(uploadServiceFacade, times(draftPosts.size + draftPages.size)).uploadPost(
                 context = any(),
                 post = any(),
                 trackAnalytics = any()
         )
     }
 
     @Test
-    fun `when uploading a single site, only posts of that site are uploaded`() {
+    fun `when uploading a single site, only posts&pages of that site are uploaded`() {
```
Comment (jd-alexander) [inline WordPress/src/test/java/org/wordpress/android/ui/uploads/UploadStarterTest.kt]:
```suggestion
    fun `when uploading a single site, only posts & pages of that site are uploaded`() {
```

File: WordPress/src/test/java/org/wordpress/android/ui/uploads/UploadStarterTest.kt [hunk 4]
```diff
@@ -188,7 +220,7 @@ class UploadStarterTest {
     }
 
     @Test
-    fun `when uploading, it ignores locally changed posts that are not publishable`() {
+    fun `when uploading, it ignores locally changed posts&pages that are not publishable`() {
```
Comment (jd-alexander) [inline WordPress/src/test/java/org/wordpress/android/ui/uploads/UploadStarterTest.kt]:
```suggestion
    fun `when uploading, it ignores locally changed posts & pages that are not publishable`() {
```

**Comment by malinajirka** [comment]
Thanks for the review @jd-alexander!


>I checked to see if the event would show up in tracks but I didn't see it but I think it might be something I am doing because I haven't used tracks alot.

It usually takes some time before the events show up in Tracks. I usually check just LogCat. If you don't see the events even in LogCat, that usually means you are opted-out (App Settings -> Privacy Settings -> Collect information).

>I requested some minor typo changes. 

I've fixed the issues in https://github.com/wordpress-mobile/WordPress-Android/pull/11157/commits/9a81d9c45a4760e1afd54f19b9341299b12a0131. 


It's ready for another round/merge. Thank you!

**Comment by jd-alexander** [comment]
@malinajirka so while testing I realized if I press the "Publish Now" button on a page in the pages list it will say "no network available" instead of moving the page to the "Published" tab similarly to how it's done when "Publish" is pressed when the page is in the editor. Is this behavior expected? Meaning, will this be addressed in another PR since this PR is simply re-enabling the feature that was disabled or should this network check be removed? Let me know 😄

**Comment by shiki** [comment]
Just did a quick glance. The changes look like what we expect (and what we reverted a long time ago). 👍 Leaving it up to you to get this merged. 🙂

**Comment by malinajirka** [comment]
@jd-alexander Thanks for the review and thorough tests;). I was aware we'd need to update all the snackbar messages, but I forgot to create an issue and estimate it :X. I've created an issue now - https://github.com/wordpress-mobile/WordPress-Android/issues/11177. I think we can merge this PR since as you mentioned it wasn't goal of this task.

@shiki Thanks for checking this ;)!

**Comment by jd-alexander** [comment]
Awesome LGTM 🚀🚀🚀🚀

**Pull request merged**